src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java

```diff
@@ -370,11 +377,16 @@ public List<ScheduledEvent> getBookingVisits(final Long bookingId, final LocalDa
     public Page<VisitWithVisitors> getBookingVisitsWithVisitor(final VisitInformationFilter filter, final Pageable pageable) {
         checkState(filter.getBookingId() != null, "BookingId required");
         final var visits = visitInformationRepository.findAll(filter, pageable);
+        final var allVisitors = getVisitorsForAllVisits(visits.getContent());
+        final var allContacts = getVisitorRelationshipsForAllVisits(filter.getBookingId(), allVisitors);
 
```

Comment:
Load all visitors and their related contacts up front for each of the visits (off which there will be up to 20)

The VisitVisitor does not model the Contact person, just the person hence the contacts loaded separately.